Build failure: open-policy-agent on aarch64-linux

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

nix-build -A open-policy-agent

bumped in https://github.com/NixOS/nixpkgs/pull/385970

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/292715867

### Relevant log output

```console
2025/03/15 12:35:05 http: TLS handshake error from 127.0.0.1:51500: remote error: tls: bad certificate
2025/03/15 12:35:05 http: TLS handshake error from 127.0.0.1:51508: remote error: tls: bad certificate
2025/03/15 12:35:05 http: TLS handshake error from 127.0.0.1:51522: remote error: tls: bad certificate
2025/03/15 12:35:05 http: TLS handshake error from 127.0.0.1:51526: remote error: tls: bad certificate
2025/03/15 12:35:05 http: TLS handshake error from 127.0.0.1:48556: read tcp 127.0.0.1:40513->127.0.0.1:48556: use of closed network connection
--- FAIL: TestInterQueryCache_ClientError (0.00s)
    --- FAIL: TestInterQueryCache_ClientError/no_raised_errors (1.16s)
        http_test.go:3447: Expected exactly 1 call to HTTP server, got 0
```

### Additional context

_No response_

### System metadata

see hydra log

### Notify maintainers

@nlewo @06kellyjac 

CC: @SuperSandro2000 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/op/open-policy-agent/package.nix
+++ b/pkgs/by-name/op/open-policy-agent/package.nix
@@ -43,9 +43,24 @@ buildGoModule rec {
     ) "opa_wasm"
   );
 
-  checkFlags = lib.optionals (!enableWasmEval) [
-    "-skip=TestRegoTargetWasmAndTargetPluginDisablesIndexingTopdownStages"
-  ];
+  checkFlags =
+    let
+      skippedTests =
+        [
+          # Skip tests that require network, not available in the nix sandbox
+          "TestInterQueryCache_ClientError"
+          "TestIntraQueryCache_ClientError"
+        ]
+        ++ lib.optionals stdenv.hostPlatform.isDarwin [
+          # Skip tests that require network, not available in the darwin sandbox
+          "TestHTTPSClient"
+          "TestHTTPSNoClientCerts"
+        ]
+        ++ lib.optionals (!enableWasmEval) [
+          "TestRegoTargetWasmAndTargetPluginDisablesIndexingTopdownStages"
+        ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
 
   preCheck =
     ''

--- a/pkgs/by-name/op/open-policy-agent/package.nix
+++ b/pkgs/by-name/op/open-policy-agent/package.nix
@@ -99,6 +99,9 @@ buildGoModule rec {
     runHook postInstallCheck
   '';
 
+  # Required for tests that need networking
+  __darwinAllowLocalNetworking = true;
+
   meta = with lib; {
     mainProgram = "opa";
     homepage = "https://www.openpolicyagent.org";

--- a/pkgs/by-name/op/open-policy-agent/package.nix
+++ b/pkgs/by-name/op/open-policy-agent/package.nix
@@ -12,14 +12,14 @@ assert
   enableWasmEval && stdenv.hostPlatform.isDarwin
   -> builtins.throw "building with wasm on darwin is failing in nixpkgs";
 
-buildGoModule rec {
+buildGoModule (finalAttrs: {
   pname = "open-policy-agent";
   version = "1.3.0";
 
   src = fetchFromGitHub {
     owner = "open-policy-agent";
     repo = "opa";
-    rev = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-wWxWpJSDOaZLJ7ULdAzPFJ9YNXX3FyQRod2roaLsuis=";
   };
 
@@ -32,7 +32,7 @@ buildGoModule rec {
   ldflags = [
     "-s"
     "-w"
-    "-X github.com/open-policy-agent/opa/version.Version=${version}"
+    "-X github.com/open-policy-agent/opa/version.Version=${finalAttrs.version}"
   ];
 
   tags = lib.optional enableWasmEval (
@@ -63,17 +63,17 @@ buildGoModule rec {
     [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
 
   preCheck =
+    # Feed in all but the e2e tests for testing
+    # This is because subPackages above limits what is built to just what we
+    # want but also limits the tests
+    # Also avoid wasm tests on darwin due to wasmtime-go build issues
     ''
-      # Feed in all but the e2e tests for testing
-      # This is because subPackages above limits what is built to just what we
-      # want but also limits the tests
-      # Also avoid wasm tests on darwin due to wasmtime-go build issues
       getGoDirs() {
         go list ./... | grep -v -e e2e ${lib.optionalString stdenv.hostPlatform.isDarwin "-e wasm"}
       }
     ''
+    # remove tests that have "too many open files"/"no space left on device" issues on darwin in hydra
     + lib.optionalString stdenv.hostPlatform.isDarwin ''
-      # remove tests that have "too many open files"/"no space left on device" issues on darwin in hydra
       rm v1/server/server_test.go
     '';
 
@@ -89,7 +89,7 @@ buildGoModule rec {
     runHook preInstallCheck
 
     $out/bin/opa --help
-    $out/bin/opa version | grep "Version: ${version}"
+    $out/bin/opa version | grep "Version: ${finalAttrs.version}"
 
     ${lib.optionalString enableWasmEval ''
       # If wasm is enabled verify it works
@@ -102,10 +102,10 @@ buildGoModule rec {
   # Required for tests that need networking
   __darwinAllowLocalNetworking = true;
 
-  meta = with lib; {
+  meta = {
     mainProgram = "opa";
     homepage = "https://www.openpolicyagent.org";
-    changelog = "https://github.com/open-policy-agent/opa/blob/v${version}/CHANGELOG.md";
+    changelog = "https://github.com/open-policy-agent/opa/blob/v${finalAttrs.version}/CHANGELOG.md";
     description = "General-purpose policy engine";
     longDescription = ''
       The Open Policy Agent (OPA, pronounced "oh-pa") is an open source, general-purpose policy engine that unifies
@@ -113,10 +113,10 @@ buildGoModule rec {
       as code and simple APIs to offload policy decision-making from your software. You can use OPA to enforce policies
       in microservices, Kubernetes, CI/CD pipelines, API gateways, and more.
     '';
-    license = licenses.asl20;
-    maintainers = with maintainers; [
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [
       lewo
       jk
     ];
   };
-}
+})

